doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -766,58 +726,81 @@ the conversion in hypotheses :n:`{+ @ident}`.
          pattern (0 = 0).
          fold not.
 
-   .. tacv:: fold {+ @term}
+.. tacn:: pattern {+, @pattern_occs } {? @occurrences }
 
-      Equivalent to :n:`fold @term ; ... ; fold @term`.
+   .. todo PR: how do multiple pattern_occs work with @occurrences??
 
-.. tacn:: pattern @term
-   :name: pattern
+   Performs :term:`eta-expansion`.  If :n:`@occurrences` is given,
+   the specified occurrences will be replaced in the selected hypotheses and/or goal.
+   If :n:`@occurrences` is not given, every occurrence of the resulting term in the goal is
+   of the current goal (say :g:`T`)
+   for a set of :n:`@one_term`\s in the :n:`@pattern_occs` by:
 
-   This command applies to any goal. The argument :n:`@term` must be a free
-   subterm of the current goal. The command pattern performs :math:`\beta`-expansion
-   (the inverse of :math:`\beta`-reduction) of the current goal (say :g:`T`) by
+   + replacing all selected occurrences of the :n:`@one_term`\s in :g:`T` with fresh variables
+   + abstracting these variables
+   + applying the abstracted goal to the :n:`@one_term`\s
 
-   + replacing all occurrences of :n:`@term` in :g:`T` with a fresh variable
-   + abstracting this variable
-   + applying the abstracted goal to :n:`@term`
+   The :n:`@one_term`\s must be free subterms of the current goal.
+
+   Starting from a goal :n:`G(t__1, …, t__n)`, :tacn:`pattern`
```

Comment:
Note that `G(t__1, …, t__n)` should be math, not Gallina.  We are talking about mathematical function application here, not an application node in the syntax of the goal.

Reply:
I think whether it's `:math:` or `:n:` is not the issue--right?  I tried to update the text to  account for `pattern` taking multiple `one_term`s.  Perhaps I blew it.  What should this be?

Reply:
The original text made it clear that the goal did not have to be headed by an application of a function to variables.  I'm suggesting that the new text souls perhaps also make this clear in some way.  (My concrete suggestion would be to stick to varphi and leave in the text about notation and substitution, but if you deliberately removed it, that's okay, and doesn't make this paragraph wrong; I'm just worried someone might be confused and assume this means the goal has to be an identifier applied to arguments)

Reply:
Deliberate to change \varphi to just φ.  Not sure what you're referring to with "leave in the text about notation and substitution".  I don't fully understand what the tactic--and happy to leave it that way for now.